src/reducers/application.js

```diff
@@ -38,31 +38,38 @@ export default (state = initialState, action) => {
       };
     }
     case SET_MESSAGE: {
+      // This causes a new messages object to be created, instead of
+      // maintaining a reference to the old data structure.
+      const messages = state.messages.map(message => message);
```

Comment:
Can we do some `...` operator magic here?

Reply:
Interesting idea!

Reply:
:+1: